### PR TITLE
outsourced a derivation of IsHomSetInhabited from Toposes to FinSetsForCAP

### PR DESCRIPTION
**Pull request by mohamed-barakat**

### CODE REVIEW

**Comment by mohamed-barakat** [comment]
@zickgraf: I am unable to write the correct logic template.

File: gap/CompilerLogic.gi
Comment (zickgraf) [inline gap/CompilerLogic.gi:417]: > @zickgraf: I am unable to write the correct logic template.

Try `not number2 ^ number1 = 0` here. For Julia, we rewrite `not a = b` to `a <> b`, but only in the output. So what you see in the output might actually not be what the syntax tree stores internally. Feel free to open an issue about this. I came across this myself recently and this certainly should be improved.

Comment (mohamed-barakat) [reply gap/CompilerLogic.gi:417]: I have already tried and it did not work.

**Comment by mohamed-barakat** [comment]
The outsourced derivation still exists in `Toposes`, which is the cause for the incomplete coverage. If I remove the derivation from `Toposes` before the PR is merged I will get an error in the CI of `CategoricalTowers`.

**Comment by mohamed-barakat** [comment]
I do not understand why the CI does not show full coverage. `IsHomSetInhabited` is covered in the examples.

File: gap/CompilerLogic.gi
Comment (zickgraf) [inline gap/CompilerLogic.gi:437]: Please uncomment this line.

File: gap/SkeletalFinSets.gi
Comment (zickgraf) [inline gap/SkeletalFinSets.gi:982]: Could you insert a description here?

**Review by zickgraf — CHANGES_REQUESTED** [review]

File: gap/CompilerLogic.gi
Comment (mohamed-barakat) [reply gap/CompilerLogic.gi]: Done.

File: gap/SkeletalFinSets.gi
Comment (mohamed-barakat) [reply gap/SkeletalFinSets.gi]: Done.

File: gap/SkeletalFinSets.gi
Comment (zickgraf) [inline gap/SkeletalFinSets.gi:982]: General comment which does not block the PR: The information you provide in this description is nearly the same as would be displayed by `DerivationsOfMethodByCategory` anyway. Unfortunately, this is the case for many descriptions of derivations. It would be more useful if descriptions would describe the mathematics/semantics behind the derivations.

**Review by zickgraf — APPROVED** [review]
Feel free to merge the PR yourself in coordination with a suitable PR to CategoricalTowers. Please check the code coverage after all PRs are merged.

**Pull request merged**
